src/ape/api/query.py

```diff
@@ -2,19 +2,37 @@
 
 import pandas as pd
 from ethpm_types.abi import EventABI, MethodABI
-from pydantic import BaseModel, NonNegativeInt, root_validator
+from pydantic import BaseModel, NonNegativeInt, root_validator, validator
 
-from ape._compat import Literal
 from ape.types import AddressType
 from ape.utils import BaseInterfaceModel, abstractmethod
 
+from .providers import BlockAPI
+from .transactions import TransactionAPI
+
 QueryType = Union["BlockQuery", "AccountQuery", "ContractEventQuery", "ContractMethodQuery"]
 
 
 class _BaseQuery(BaseModel):
-    type: str  # Used as discriminator
+
     columns: List[str]
 
+    @classmethod
+    @abstractmethod
+    def all_fields(cls) -> List[str]:
+        ...
+
+    @validator("columns")
+    def check_columns(cls, data):
+        all_fields = cls.all_fields()
+        if len(data) == 1 and data[0] == "*":
+            return all_fields
+        else:
+            assert len(set(data)) == len(data), f"Duplicate fields in {data}"
```

Comment:
NOTE: This assertions get removed if running Python outside of debug mode. Are you sure that is what we want? Otherwise, just raise `AssertionError`s manually

Reply:
Ah yes, raise `ValueError` instead here to match typical pydantic validators